concourse/verify-production-urls/script.bash

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -ex
+
+production_url=https://openstax.org
+
+work_dir=$(pwd)
+
+release_id=$(curl "${production_url}/rex/environment.json" | jq .release_id -r)
+files=$(aws s3api list-objects --bucket "$PROD_UNIFIED_S3_BUCKET" --prefix "rex/releases/$release_id/books/" | jq -r '.Contents[] | .Key')
+
+missing_files=""
+
+for full_path in $files; do
+  book_path="/release/$(grep -oP 'books.*' <<< "$full_path")"
+  [ ! -f "$work_dir$book_path" ] && missing_files="$missing_files$book_path'\n'"
```

Comment:
you don't have to collect `missing_files`, you can `echo` them as you go. but you do have to track if anything fails so you can `exit` with the right code. the way you have it now the script will always register as passing even if it finds issues. check out `concourse/execute-test-runs/script.bash ` for an example of how it stores a `failed` variable and then does `exit "$failed"` to properly succeed or fail based on multiple commands